### ISSUE TITLE
Protected fields in public stores

### ISSUE THREAD

**Issue by Schwartz10** [3box/3box-js#415]
**FEATURE Request**

It would be great to have some added `type` protection for more important and frequently used fields in a 3box. For instance, I don't want another dapp to accidentally set my user's `name` field to an `object` - thus breaking all dapps that are properly using `name`. 

Not sure the best way to handle this. But one thought I had was to create another call to create a `protectedField` which comes with a `type`. And if someone else tries to `setProtectedField` with the wrong `type`, it would throw a helpful error message. All the current 3box fields should be considered "protected"

**Comment by oed** [comment]
I think something along these lines makes a lot of sense. 
Would be nice if when creating a space a CID of a schema object could be passed along. The store would then resolve this object and use it to enforce protected fields. This is something that would need to be implemented in the orbit-db-store layer anyhow, in order for it to be enforceable.

**Comment by Schwartz10** [comment]
Yeah, i think using JSON schema would be perfect for this https://json-schema.org/understanding-json-schema/
